tests/test_fields.py

```diff
@@ -128,6 +128,17 @@ def test_model_field_convert_to_type_unset(typeless_model_field):
     assert typeless_model_field.convert_to_type(None, value, field_class=typing.Any) is value
 
 
+@pytest.mark.parametrize('value', (1, '1'))
+def test_model_field_convert_to_type_union(typeless_model_field, value):
+    assert typeless_model_field.convert_to_type(None, value, field_class=typing.Union[int, str]) is value
+
+
+def test_model_field_convert_to_type_union_invalid(typeless_model_field):
+    value = dict()
+    with pytest.raises(AssertionError):
```

Comment:
I think we should check the message error just to be sure the wrong `AssertionError` isn't being captured